plugins/providers/hyperv/scripts/import_vm_vmcx.ps1

```diff
@@ -0,0 +1,154 @@
+
+$VmProperties = @{
+    Path = $vm_config_file 
+    SnapshotFilePath   = Join-Path $data_path 'Snapshots'
+    VhdDestinationPath = Join-Path $data_path 'Virtual Hard Disks'
+    VirtualMachinePath = $data_path
+}
+
+$vmConfig = (Compare-VM -Copy -GenerateNewID @VmProperties)
+
+$generation = $vmConfig.VM.Generation
+
+if (!$vmname) {
+    # Get the name of the vm
+    $vm_name = $vmconfig.VM.VMName
+}else {
+    $vm_name = $vmname
+}
+
+if (!$cpus) {
+    # Get the processorcount of the VM
+    $processors = (Get-VMProcessor -VM $vmConfig.VM).Count
+}else {
+    $processors = $cpus
+}
+
+function GetUniqueName($name) {
+    Get-VM | ForEach-Object -Process {
+        if ($name -eq $_.Name) {
+            $name =  $name + "_1"
+        }
+    }
+    return $name
+}
+
+do {
+    $name = $vm_name
+    $vm_name = GetUniqueName $name
+} while ($vm_name -ne $name)
+
+if (!$memory) {
+    $configMemory = Get-VMMemory -VM $vmConfig.VM
+    $dynamicmemory = $configMemory.DynamicMemoryEnabled
+
+    $MemoryMaximumBytes = ($configMemory.Maximum)
+    $MemoryStartupBytes = ($configMemory.Startup)
+    $MemoryMinimumBytes = ($configMemory.Minimum)
+} else {
+    if (!$maxmemory){
+        $dynamicmemory = $False
+        $MemoryMaximumBytes = ($memory -as [int]) * 1MB
+        $MemoryStartupBytes = ($memory -as [int]) * 1MB
+        $MemoryMinimumBytes = ($memory -as [int]) * 1MB
+    } else {
+        $dynamicmemory = $True
+        $MemoryMaximumBytes = ($maxmemory -as [int]) * 1MB
+        $MemoryStartupBytes = ($memory -as [int]) * 1MB
+        $MemoryMinimumBytes = ($memory -as [int]) * 1MB
+    }
+}
+
+if (!$switchname) {
+    $switchname = (Get-VMNetworkAdapter -VM $vmConfig.VM).SwitchName
+}
+
+Connect-VMNetworkAdapter -VMNetworkAdapter (Get-VMNetworkAdapter -VM $vmConfig.VM) -SwitchName $switchname
+Set-VM -VM $vmConfig.VM -NewVMName $vm_name -MemoryStartupBytes $MemoryStartupBytes
+Set-VM -VM $vmConfig.VM -ErrorAction "Stop" -ProcessorCount $processors
+
+If ($dynamicmemory) {
+    Set-VM -VM $vmConfig.VM -DynamicMemory
+    Set-VM -VM $vmConfig.VM -MemoryMinimumBytes $MemoryMinimumBytes -MemoryMaximumBytes $MemoryMaximumBytes
+} else {
+    Set-VM -VM $vmConfig.VM -StaticMemory    
+}
+
+if ($notes) {
+    Set-VM -VM $vmConfig.VM -Notes $notes
+}
+
+if ($auto_start_action) {
+    Set-VM -VM $vmConfig.VM -AutomaticStartAction $auto_start_action
+}
+
+if ($auto_stop_action) {
+    Set-VM -VM $vmConfig.VM -AutomaticStartAction $auto_stop_action
+}
+
+# Only set EFI secure boot for Gen 2 machines, not gen 1
+if ($generation -ne 1) {
+    Set-VMFirmware -VM $vmConfig.VM -EnableSecureBoot (Get-VMFirmware -VM $vmConfig.VM).SecureBoot
+}
+
+$report = Compare-VM -CompatibilityReport $vmConfig
+
+# Stop if there are incompatibilities
+if($report.Incompatibilities.Length -gt 0){
+    Write-Error-Message $(ConvertTo-Json $($report.Incompatibilities | Select -ExpandProperty Message))
+    exit 0
```

Comment:
Hmm, sematicly, you are correct. But what i understood from how the code of Vagrant works it just parses the output and handles the error and prints the error in a nice way. Think the default error handling is a bit more rough in its messages. 

That is how i understood the code at least :)

Relevant code: https://github.com/bbrala/vagrant/blob/8744caebcf50095e808233fd7282849115eddacb/plugins/providers/hyperv/driver.rb#L20-L40